pkg/webhook/webhook.go

```diff
@@ -122,10 +143,12 @@ func (wh *WebHook) Start(webhookConfigName string) error {
 
 // Stop deregisters itself with the API server and stops the admission webhook server.
 func (wh *WebHook) Stop(webhookConfigName string) error {
-	if err := wh.selfDeregistration(webhookConfigName); err != nil {
-		return err
+	if wh.failurePolicy != arv1beta1.Fail {
```

Comment:
Can you explain why this change is needed?

Reply:
I'll add a comment. If the webhook deregisters itself, some pods may be scheduled without hitting the webhook (for example, if the webhook gets deregistered while a driver is creating executors).